app/code/Codifi/Sales/Plugin/Magento/Sales/Block/Adminhtml/Reorder/Renderer/ActionPlugin.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Codifi_Sales
+ *
+ * @copyright   Copyright (c) 2021 Codifi
+ * @author      Pavel Zelenevich <pzelenevich@codifi.me>
+ */
+
+namespace Codifi\Sales\Plugin\Magento\Sales\Block\Adminhtml\Reorder\Renderer;
+
+use Magento\Sales\Model\OrderRepository;
+use Magento\Sales\Block\Adminhtml\Reorder\Renderer\Action;
+use Magento\Framework\Exception\InputException;
+use Magento\Framework\Exception\NoSuchEntityException;
+
+/**
+ * Class ActionPlugin
+ * @package Codifi\Sales\Plugin\Magento\Sales\Block\Adminhtml\Reorder\Renderer
+ */
+class ActionPlugin
+{
+    /**
+     * Order repository
+     *
+     * @var OrderRepository
+     */
+    private $orderRepository;
+
+    /**
+     * ActionPlugin constructor.
+     * @param OrderRepository $orderRepository
+     */
+    public function __construct(OrderRepository $orderRepository)
+    {
+        $this->orderRepository = $orderRepository;
+    }
+
+    /**
+     * Hide reorder link if order_type - credit hold
+     *
+     * @param Action $subject
+     * @param callable $proceed
+     * @param array $actionArray
+     * @return null
+     * @throws InputException
+     * @throws NoSuchEntityException
+     */
+    public function aroundAddToActions(Action $subject, callable $proceed, array $actionArray)
+    {
+        $urlPath = parse_url($actionArray["@"]['href'], PHP_URL_PATH);
```

Comment:
It's better to move this logic in new method and before get order check that order_id is found
```
$orderId = $this->getOrderIdParamFromUrl($url);
if ($orderId) {
...
}
```